core/op_s3.go

```diff
@@ -3,364 +3,193 @@ package core
-	if err != nil {
-		return opType, jobResponse(err)
+		err = client.Delete(wp.ctx, src)
 	}
 
-	err = wildOperation(client, src.url, true, wp, func(object *storage.Object) *Job {
-		if object.Mode.IsDir() {
-			return nil
-		}
-
-		return makeJob(object)
-	})
-
-	return opType, jobResponse(err)
+	return jobResponse(err)
 }
 
-func S3BatchDeleteActual(job *Job, wp *WorkerParams) (stats.StatType, *JobResponse) {
-	const opType = stats.S3Op
-
-	src := job.args[0]
-
-	client, err := wp.newClient(src.url)
-	if err != nil {
-		return opType, jobResponse(err)
-	}
-
-	deleteObjects := make([]*objurl.ObjectURL, len(job.args)-1)
-	for i, a := range job.args {
-		if i == 0 {
-			continue
-		}
-		deleteObjects[i-1] = a.url
-	}
+func S3Delete(job *Job, wp *WorkerParams) *JobResponse {
+	src := job.src[0]
 
-	err = client.Delete(wp.ctx, deleteObjects...)
+	client, err := wp.newClient(src)
 	if err != nil {
-		return opType, jobResponse(err)
+		return jobResponse(err)
 	}
 
-	st := client.Statistics()
-
-	var msg []string
-	for key, stat := range st.Keys() {
-		if stat.Success {
-			msg = append(msg, fmt.Sprintf("Batch-delete %v", key))
-		} else {
-			msg = append(msg, fmt.Sprintf(`Batch-delete %v: %s`, key, stat.Message))
-		}
-	}
-
-	return opType, jobResponse(err, msg...)
+	err = client.Delete(wp.ctx, src)
+	return jobResponse(err)
 }
 
-func S3BatchDownload(job *Job, wp *WorkerParams) (stats.StatType, *JobResponse) {
-	const opType = stats.S3Op
-
-	subCmd := "cp"
-	if job.operation == op.AliasBatchGet {
-		subCmd = "get"
-	}
+func S3Download(job *Job, wp *WorkerParams) *JobResponse {
+	src, dst := job.src[0], job.dst
 
-	if job.opts.Has(opt.DeleteSource) {
-		subCmd = "mv"
+	response := CheckConditions(src, dst, wp, job.opts)
+	if response != nil {
+		return response
 	}
-	subCmd += job.opts.GetParams()
 
-	src, dst := job.args[0], job.args[1]
-
-	client, err := wp.newClient(src.url)
+	client, err := wp.newClient(src)
 	if err != nil {
-		return opType, jobResponse(err)
+		return jobResponse(err)
 	}
-	err = wildOperation(client, src.url, true, wp, func(object *storage.Object) *Job {
-		if object.IsMarker() || object.Mode.IsDir() {
-			return nil
-		}
-
-		arg1 := NewJobArgument(object.URL)
-
-		var dstFn string
-		if job.opts.Has(opt.Parents) {
-			dstFn = object.URL.Path
-		} else {
-			dstFn = object.URL.Base()
-		}
-
-		arg2 := dst.Join(dstFn)
-
-		subJob := job.MakeSubJob(subCmd, op.Download, []*JobArgument{arg1, arg2}, job.opts)
-
-		dir := filepath.Dir(arg2.url.Absolute())
-		os.MkdirAll(dir, os.ModePerm)
-		return subJob
-	})
-
-	return opType, jobResponse(err)
-}
-
-func S3Download(job *Job, wp *WorkerParams) (stats.StatType, *JobResponse) {
-	const opType = stats.S3Op
 
-	src, dst := job.args[0], job.args[1]
-
-	response := CheckConditions(src, dst, wp, job.opts)
-	if response != nil {
-		return opType, response
-	}
-
-	srcFn := src.url.Base()
-	destFn := dst.url.Absolute()
+	srcFn := src.Base()
+	destFn := dst.Absolute()
```

Comment:
while we are here, should we change `Fn` convention to more readable variable names? dstFilename or similar.